### PR TITLE
cdc/kafka: remove skip-auto-create failpoint in kafka test

### LINKED ISSUE #4660: remove `SkipAutoCreate` failpoint from release-5.3

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #4660

### What is changed and how it works?

port `ClusterAdminClientMockImpl` from master, with all necessary changes to test code 

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: cdc/sink/producer/kafka/kafka.go [hunk 1]
```diff
@@ -660,6 +579,96 @@ func kafkaClientID(role, captureAddr, changefeedID, configuredClientID string) (
 	return
 }
 
+func validateAndCreateTopic(admin kafka.ClusterAdminClient, topic string, config *Config, saramaConfig *sarama.Config) error {
```
Comment (asddongmen) [inline cdc/sink/producer/kafka/kafka.go:582]: The name `admin` is confusing.

**Review by hi-rustin — APPROVED** [review]

File: cdc/sink/producer/kafka/kafka.go [hunk 2]
```diff
@@ -660,6 +579,96 @@ func kafkaClientID(role, captureAddr, changefeedID, configuredClientID string) (
...
+		if err != nil {
+			return cerror.WrapError(cerror.ErrKafkaNewSaramaProducer, err)
+		}
+		topicMaxMessageBytes, err := strconv.Atoi(topicMaxMessageBytesStr)
+		if err != nil {
+			return cerror.WrapError(cerror.ErrKafkaNewSaramaProducer, err)
+		}
+
+		if topicMaxMessageBytes < config.MaxMessageBytes {
+			log.Warn("topic's `max.message.bytes` less than the `max-message-bytes`,"+
```
Comment (asddongmen) [inline cdc/sink/producer/kafka/kafka.go:603]: the existing topic's `max.message.bytes` less than the `max-message-bytes` specified in config

**Review by asddongmen — APPROVED** [review]

**Pull request merged**